src/runner.ts

```diff
@@ -1,38 +1,29 @@
-import { setOutput, summary } from "@actions/core";
+import { setOutput } from "@actions/core";
 import { parse } from "yaml";
 
 import { Inputs } from ".";
+import {
+  CommonRuleFailure,
+  FellowMissingRankFailure,
+  RequiredReviewersData,
+  ReviewFailure,
+  RuleFailedReport,
+  RuleFailedSummary,
+} from "./failures";
 import { GitHubChecksApi } from "./github/check";
 import { PullRequestApi } from "./github/pullRequest";
 import { ActionLogger, CheckData, TeamApi } from "./github/types";
 import { AndDistinctRule, ConfigurationFile, FellowsRule, Reviewers, Rule, RuleTypes } from "./rules/types";
 import { validateConfig, validateRegularExpressions } from "./rules/validator";
-import { caseInsensitiveEqual, concatArraysUniquely } from "./util";
-
-type ReviewReport = {
-  /** The amount of missing reviews to fulfill the requirements */
-  missingReviews: number;
-  /** The users who would qualify to complete those reviews */
-  missingUsers: string[];
-  /** If applicable, the teams that should be requested to review */
-  teamsToRequest?: string[];
-  /** If applicable, the users that should be requested to review */
-  usersToRequest?: string[];
-  /** If applicable, the missing minimum fellows rank required to review */
-  missingRank?: number;
-  /** If applicable, reviews that count towards this rule */
-  countingReviews: string[];
-};
```

Comment:
This piece has been moved to the `failure/types.ts` file and split into smaller sections.